services/DeviceInformation/include/ble-service-device-information/DeviceInformationService.h

```diff
@@ -0,0 +1,110 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef BLE_DEVICE_INFORMATION_SERVICE_H
+#define BLE_DEVICE_INFORMATION_SERVICE_H
+
+#include "ble/BLE.h"
+
+#if BLE_FEATURE_GATT_SERVER
+
+/** The Device Information Service exposes manufacturer and/or vendor information about a device.
+ *
+ * The characteristics added are read only and written once. Do no construct this class.
+ * Use the static method add_service to add the chosen Device Information Service characteristics to the server.
+ *
+ * You can read the specification of the service on the bluetooth website, currently at:
+ * https://www.bluetooth.com/specifications/specs/
+ * Otherwise search the website for "Device Information Service".
+ */
+class DeviceInformationService {
+public:
+    /** System ID */
+    struct system_id_t {
+        /** Bottom 5 bytes used. */
+        uint32_t organizationally_unique_identifier;
+        /** Bottom 3 bytes used. */
+        uint64_t manufacturer_defined_identifier;
+    };
+
+    /** Plug and Play ID */
+    struct pnp_id_t {
+        /** Specifies where the id is taken from. Currently the only legal values are:
+         * 0x01 Bluetooth SIG-assigned Device ID Vendor ID value from the Assigned Numbers document,
+         * 0x02 USB Implementer’s Forum assigned Vendor ID value.
+         * Other values reserved for future use. */
+        uint8_t 	vendor_id_source;
+        /** The Vendor IDfield is intended to uniquely identify the vendor of the device. */
+        uint16_t 	vendor_id;
+        /** The ProductID field is intended to distinguish between different products made by the vendor
+         * identified with the Vendor ID field. The vendors themselves manage Product ID field values. */
+        uint16_t 	product_id;
+        /** The value of the field follows the scheme 0xJJMN for version JJ.M.N
+         * (JJ – major version number, M – minor version number, N – sub-minor version number). */
+        uint16_t 	product_version;
+    };
+
+    /** IEEE 11073-20601 Regulatory Certification Data List */
+    struct regulatory_cert_data_list_t {
+        /** Regulatory and certification information for the product in a list defined in IEEE 11073-20601.
+         *
+         * Must conform to IEEE Std 11073-20601-2008 Health Informatics - Personal Health Device Communication
+         * Application Profile - Optimized Exchange Protocol version 1.0 or later.
+         *
+         * @Attention Pointer must be valid and first byte must be equal to the size of the data
+         * (one byte less than size of the buffer). */
+        uint8_t *data;
```

Comment:
I think we should make it const otherwise it will be difficult to link with variable in the text segment: 

```c++
static constexpr const uint8_t reg_cert_data_list[] = { 
    2, 0xDE, 0xAD
};

void foo() { 
    regulatory_cert_data_list_t reg_cert;
    reg_cert.data = reg_cert_data_list;    /// Compilation error if data is not const!
}
```